src/Core/Configurations/ConfigFileWatcher.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Core.Configurations;
+
+public class ConfigFileWatcher
+{
+    private FileSystemWatcher? _fileWatcher;
+    private readonly RuntimeConfigProvider? _configProvider;
+
+    public ConfigFileWatcher(RuntimeConfigProvider configProvider, string directoryName, string configFileName)
+    {
+        _fileWatcher = new FileSystemWatcher(directoryName)
+        {
+            Filter = configFileName,
+            EnableRaisingEvents = true
+        };
+
+        _configProvider = configProvider;
+        _fileWatcher.Changed += OnConfigFileChange;
+    }
+
+    public ConfigFileWatcher() { }
```

Comment:
we simply use this to avoid instantiating the actual FileWatching mechanic but being able to check if we have already setup this wrapper object for cases like lateconfig, let's us avoid tracking this information by just making a null check..

Reply:
that would be another way to do this, but then you're running that logic every single time you try to get config, this let's us just see if the object is null or not and then skip all of that logic.

Reply:
updated to use bool for beauty.

Reply:
talked offline and updated to not create empty watcher object.

Reply:
that may have been missed. @aaronburtle, you'll need to remove this now.